### PR TITLE
[SPARK-12716] [Web UI] Add a TOTALS row to the Executors Web UI

### PR DESCRIPTION
**Pull request by ajbozarth**
Added a Totals table to the top of the page to display the totals of each applicable column in the executors table.

Old Description:
~~Created a TOTALS row containing the totals of each column in the executors UI. By default the TOTALS row appears at the top of the table. When a column is sorted the TOTALS row will always sort to either the top or bottom of the table.~~

### CODE REVIEW

**Comment by ajbozarth** [comment]
Screenshots:
Initial page load
initial
Sort by address
sortaddrbottom
sortaddrtop
Sort by ID
sortidbottom
sortidtop
Sort by Task Count
sorttasksbottom
sorttaskstop

**Comment by ajbozarth** [comment]
@srowen @tgravescs This is for the second half of SPARK-1832 that you reported if you want to take a look

**Comment by tgravescs** [comment]
thanks for working on this.  After seeing it I'm wondering if it would look better as a separate table just for the summary.  Similar to the "Summary Metrics for X Completed Tasks" for a stage.

what do you guys think? copy @JoshRosen

**Comment by ajbozarth** [comment]
@tgravescs I can put together an implementation of that to compare to, I don't know which would look better though, The stages summary doesn't have the same fields as the tables below it like this would, might look odd.

**Comment by ajbozarth** [comment]
Here's a screenshot of @tgravescs summary table idea. [Note: this code is NOT committed] What do people think of this alternative? Opinions on how to improve it? I think we need to add a subhead for the original table, but I'm unsure of what it should say
screen shot 2016-01-11 at 4 27 11 pm

**Comment by tgravescs** [comment]
personally I like this better because its always there and not moving when sorting. 

One other reason I was going for a separate table is I filed another jira to show killed executors.  I was thinking it might work as a nice summary for that where you could show active executors, inactive, as well as summaries across.

**Comment by ajbozarth** [comment]
I just looked at that other PR and this will look great once that get's merged, I'll update the code to match the above screenshot

**Comment by ajbozarth** [comment]
Do we want to wait for PR #10058 and then add an ACTIVE and DEAD row to the the table? Or Is this good to go?

#10058 seems stale, but I feel like the table looks off with just one row.

Also I think adding a header to the original table would look better but I personally can't think of what the header would say other than "Executors" which is already the page header, opinions?

**Comment by tgravescs** [comment]
No we don't need to wait for that.  I actually filed a new jira for what looks like the same thing.

You have "Totals for 5 Executors" on the top table so how about the bottom one we just put "Active Executors"  since it doesn't show ones that are killed anyway.

File: core/src/main/scala/org/apache/spark/ui/exec/ExecutorsPage.scala [hunk 1]
```diff
@@ -93,14 +93,10 @@ private[ui] class ExecutorsPage(
       </table>
 
     val content =
-      <div class="row-fluid">
+      <div class="row">
         <div class="span12">
-          <ul class="unstyled">
-            <li><strong>Memory:</strong>
-              {Utils.bytesToString(memUsed)} Used
```
Comment (tgravescs) [inline core/src/main/scala/org/apache/spark/ui/exec/ExecutorsPage.scala]: memUsed and maxMem locals in this function aren't being used anymore so lets remove them.

File: core/src/main/scala/org/apache/spark/ui/exec/ExecutorsPage.scala [hunk 2]
```diff
@@ -93,14 +93,10 @@ private[ui] class ExecutorsPage(
...
-              ({Utils.bytesToString(maxMem)} Total) </li>
-            <li><strong>Disk:</strong> {Utils.bytesToString(diskUsed)} Used </li>
```
Comment (tgravescs) [inline core/src/main/scala/org/apache/spark/ui/exec/ExecutorsPage.scala]: same here I diskUsed isn't be used

**Comment by ajbozarth** [comment]
screen shot 2016-01-14 at 3 10 10 pm

**Comment by tgravescs** [comment]
+1, thanks @ajbozarth

**Comment by tgravescs** [comment]
Oops sorry, it looks like it doesn't merge cleanly anymore.  can you upmerge again.

**Comment by ajbozarth** [comment]
The tests finished out of order, check the commit numbers. I thought that first too.

**Pull request closed without merging**